analytic_engine/src/table/mod.rs

```diff
@@ -112,19 +359,21 @@ impl Table for TableImpl {
     }
 
     async fn write(&self, request: WriteRequest) -> Result<usize> {
+        if self.instance.max_rows_in_write_queue > 0 {
+            return self.write_with_pending_queue(request).await;
+        }
+
         let mut serial_exec = self.table_data.serial_exec.lock().await;
```

Comment:
`try_lock` here? if failed, then `write_with_pending_queue`.

Reply:
There is no need to submit the write request to the queue if the queue's capacity is smaller than the write requests.